{"name": "test_place_order_flow", "status": "failed", "statusDetails": {"message": "AssertionError: ❌ Purchase confirmation not found within timeout.\nPage source snippet:\n<html lang=\"en\"><head><style class=\"vjs-styles-defaults\">\n      .video-js {\n        width: 300px;\n        height: 150px;\n      }\n\n      .vjs-fluid {\n        padding-top: 56.25%\n      }\n    </style><style class=\"vjs-styles-dimensions\">\n      .example-video-dimensions {\n        width: 400px;\n        height: 320px;\n      }\n\n      .example-video-dimensions.vjs-fluid {\n        padding-top: 56.25%;\n      }\n    </style>\n\n  <meta charset=\"utf-8\">\n  <meta name=\"viewport\" content=\"width=device-width, init", "trace": "self = <pages.checkout_page.CheckoutPage object at 0x00000268DB8E1D90>\n\n    def confirm_purchase(self):\n        \"\"\"Click purchase button and verify confirmation appears\"\"\"\n        self.driver.find_element(By.XPATH, PURCHASE_BUTTON).click()\n    \n        try:\n            # Wait for confirmation header or modal body text\n>           confirmation_element = WebDriverWait(self.driver, 30).until(\n                EC.any_of(\n                    EC.visibility_of_element_located((By.XPATH, CONFIRMATION_HEADER_XPATH)),\n                    EC.visibility_of_element_located((By.XPATH, \"//p[contains(text(),'Thank you')]\")),\n                    EC.visibility_of_element_located((By.XPATH, \"//h2[contains(text(),'Thank you')]\"))\n                )\n            )\n\npages\\checkout_page.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"b8c0491a0d2599eeb2b7a1b5fb990c961376050e\")>, method = <function any_of.<locals>.any_of_condition at 0x00000268DB8D8680>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        -----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\nvenv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:138: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"b8c0491a0d2599eeb2b7a1b5fb990c961376050e\")>\n\n    @allure.feature('Place Order')\n    def test_place_order_flow(driver):\n        # load test user data\n        data_file = os.path.join(os.path.dirname(__file__), 'users.csv')\n        with open(data_file, newline='') as f:\n            reader = csv.DictReader(f)\n            users = list(reader)\n        user = users[0]\n    \n        home = HomePage(driver)\n        home.open_first_product()\n    \n        product = ProductPage(driver)\n        alert_text = product.add_to_cart_and_accept_alert()\n    \n        cart = CartPage(driver)\n        cart.go_to_cart()\n        cart.place_order()\n    \n        checkout = CheckoutPage(driver)\n        checkout.fill_order(user['name'], user['country'], user['city'],\n                            user['card'], user['month'], user['year'])\n>       checkout.confirm_purchase()\n\ntests\\test_place_order.py:31: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <pages.checkout_page.CheckoutPage object at 0x00000268DB8E1D90>\n\n    def confirm_purchase(self):\n        \"\"\"Click purchase button and verify confirmation appears\"\"\"\n        self.driver.find_element(By.XPATH, PURCHASE_BUTTON).click()\n    \n        try:\n            # Wait for confirmation header or modal body text\n            confirmation_element = WebDriverWait(self.driver, 30).until(\n                EC.any_of(\n                    EC.visibility_of_element_located((By.XPATH, CONFIRMATION_HEADER_XPATH)),\n                    EC.visibility_of_element_located((By.XPATH, \"//p[contains(text(),'Thank you')]\")),\n                    EC.visibility_of_element_located((By.XPATH, \"//h2[contains(text(),'Thank you')]\"))\n                )\n            )\n            return confirmation_element.text\n        except TimeoutException:\n            # Capture page source for debugging\n            page_source = self.driver.page_source\n>           raise AssertionError(\n                \"❌ Purchase confirmation not found within timeout.\\n\"\n                \"Page source snippet:\\n\" + page_source[:500]\n            )\nE           AssertionError: ❌ Purchase confirmation not found within timeout.\nE           Page source snippet:\nE           <html lang=\"en\"><head><style class=\"vjs-styles-defaults\">\nE                 .video-js {\nE                   width: 300px;\nE                   height: 150px;\nE                 }\nE           \nE                 .vjs-fluid {\nE                   padding-top: 56.25%\nE                 }\nE               </style><style class=\"vjs-styles-dimensions\">\nE                 .example-video-dimensions {\nE                   width: 400px;\nE                   height: 320px;\nE                 }\nE           \nE                 .example-video-dimensions.vjs-fluid {\nE                   padding-top: 56.25%;\nE                 }\nE               </style>\nE           \nE             <meta charset=\"utf-8\">\nE             <meta name=\"viewport\" content=\"width=device-width, init\n\npages\\checkout_page.py:39: AssertionError"}, "attachments": [{"name": "log", "source": "8c6e510c-588f-4e69-a659-35437a13178d-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "e42edac3-ab5e-439c-b203-010fb0af5732-attachment.txt", "type": "text/plain"}], "start": 1761729180829, "stop": 1761729233096, "uuid": "e396852b-cadb-4175-ad17-9f41c1ab5a69", "historyId": "4f74172bf10135fb4afff49bfb34c0c6", "testCaseId": "4f74172bf10135fb4afff49bfb34c0c6", "fullName": "tests.test_place_order#test_place_order_flow", "labels": [{"name": "feature", "value": "Place Order"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_place_order"}, {"name": "host", "value": "ASCENDI-N7SB05U"}, {"name": "thread", "value": "13860-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_place_order"}], "titlePath": ["tests", "test_place_order.py"]}